doc/dev/design/workflows/storage_devices_and_osds.rst

```diff
@@ -0,0 +1,304 @@
+-----------------------
+
+**1. Context is everything**:
+The current OSD creation flow doesn’t provide any indications of available devices or hosts. This leaves the user clicking on the add button and seeing nothing, if there are no devices available - at which point the user assumes there are no available devices. Both host-mode and device-mode UI flows illustrate a couple of usability features that should be implemented as a bare minimum.
+
+    a. If there are no devices available, the add button should be disabled
+
+    b. The UI for OSD creation should include a summary of discovered hosts with disks and the total number of available disks that could be used for OSD creation. This should also show total raw. E.g. 5 hosts, 50 HDDs (80TB), 10 NVME (5TB)
+
+        - The discovered configuration could also
+            - Use the hosts rack ID annotations to look at the capacity from a fault domain perspective to ensure it’s balanced - and warn if not.
+            - Confirm whether the host configurations are identical (homogeneous). Heterogeneous configurations could therefore be accompanied by a INFO/WARN message in the UI to highlight the potential balance issues of heterogeneous clusters.
+
+    c. Once the deployment decision is made, display a summary of the selection, that the user CONFIRMs
+        * Total devices by type that would be used
+        * Total number of OSDs that would be created
+        * Overall raw capacity of the creation request, together with the potential raw cluster capacity once the OSD addition is complete
+        * Use a rule-of-thumb to determine approximate deployment time - set an expectation.
+
+**2. Enabling new capacity**:
+Policy option for how new disks are added to the cluster (this is present in both host-mode and device-mode designs)
+- Phased by OSD: All OSDs added are at weight 0. The orchestrator then reweights each OSD in turn to drive the rebalance
+- Phased by host: all OSDs on a given host are reweighted at the same time
+- Immediate: don’t use reweight. Bring the OSDs up/in straight away (on an empty cluster, this should be the default)
+
+**3. UI redesign**:
+Discover the devices, suggest a layout based on these devices combined with best practice, inform if there are is not enough flash for the number of HDDs, inform if there are no free devices, and also provide the advanced use case which is what we see today (which echoes the drive group approach)
+
+**4. Imperative not Declarative**:
+The use of declarative “drive groups” is a problem in several aspects:
+
+For the final user:
+
+The “admin persona” who is going to install a cluster by first time knows what is the current hardware composition and will create the OSD’s possibly using all the storage devices in the hosts of the cluster planified to harbor OSDs.
+
+But we are not telling the “admin persona” that this initial decision will be inmutable in the future and applied automatically without any warning.
```

Comment:
Agreed. Or provide a cluster-wide switch to enable/disable this auto deploy behaviour.

I also don't think this decision is immutable, though my experience has also been that it is unreasonably hard to apply a new drive group spec to a host.

Reply:
> I also don't think this decision is immutable, though my experience has also been that it is unreasonably hard to apply a new drive group spec to a host.

Is hard because you are thinking that drive group can only be applied in declarative mode, and this implies to merge the first drive group spec where this host was included with new one specs... something very difficult to implement (at least it seems to be that)